tests/unit_tests/pyanaconda_tests/modules/timezone/test_module_timezone.py

```diff
@@ -521,61 +521,49 @@ def _get_test_sources(self):
 
         return [server, pool]
 
-    def _get_expected_lines(self):
-        return [
-            "server unique.ntp.server iburst\n",
-            "pool another.unique.server\n"
-        ]
-
     def _test_ntp_inputs(self, make_chronyd=False, ntp_enabled=True, ntp_installed=False,
                          ntp_config_error=False):
         ntp_servers = self._get_test_sources()
-        expected_lines = self._get_expected_lines()
-
+        expected_lines = [
+            "server unique.ntp.server iburst\n",
+            "pool another.unique.server\n"
+        ]
         with tempfile.TemporaryDirectory() as sysroot:
             self._setup_environment(sysroot, make_chronyd)
 
-            with patch("pyanaconda.modules.timezone.installation.service") as service_util:
-                service_util.is_service_installed.return_value = ntp_installed
-                self._execute_task(sysroot, ntp_enabled, ntp_servers)
-                self._validate_ntp_service(sysroot, service_util, ntp_installed, ntp_enabled)
-
-            if ntp_config_error:
-                return
+            self._test_task(sysroot, ntp_enabled, ntp_installed, ntp_servers)
 
-            self._validate_ntp_config(sysroot, make_chronyd, ntp_enabled, expected_lines)
+            if not ntp_config_error:
+                self._validate_ntp_config(sysroot, make_chronyd, ntp_enabled, expected_lines)
 
     def _setup_environment(self, sysroot, make_chronyd):
         os.mkdir(sysroot + "/etc")
         if make_chronyd:
             copyfile(NTP_CONFIG_FILE, sysroot + NTP_CONFIG_FILE)
 
-    def _execute_task(self, sysroot, ntp_enabled, ntp_servers):
-        task = ConfigureNTPTask(
-            sysroot=sysroot,
-            ntp_enabled=ntp_enabled,
-            ntp_servers=ntp_servers
-        )
-        task.run()
-
-    def _validate_ntp_service(self, sysroot, service_util, ntp_installed, ntp_enabled):
-        service_util.is_service_installed.assert_called_once_with(
-            "chronyd", root=sysroot
-        )
-
-        if not ntp_installed:
-            service_util.enable_service.assert_not_called()
-            service_util.disable_service.assert_not_called()
-        elif ntp_enabled:
-            service_util.enable_service.assert_called_once_with(
-                "chronyd", root=sysroot
-            )
-            service_util.disable_service.assert_not_called()
-        else:
-            service_util.enable_service.assert_not_called()
-            service_util.disable_service.assert_called_once_with(
-                "chronyd", root=sysroot
-            )
+    def _test_task(self, sysroot, ntp_enabled, ntp_installed, ntp_servers):
+        PREFIX = "pyanaconda.modules.timezone.installation"
+        with patch(PREFIX + ".is_service_installed") as mock_is_installed:
+            with patch(PREFIX + ".enable_service") as mock_enable:
+                with patch(PREFIX + ".disable_service") as mock_disable:
```

Comment:
Not sure if there is a better way to do this but this is a kind of ugly :(.

Reply:
Oh, it is absolutely ugly. We discussed this part with Vendy and there are essentially three options here:

- Write a custom context manager that wraps the three patches. However that only moves the problem around, at a cost I don't want to pay.
- Give up and drop the last commit. The `module.function` call style remains.
- Nuke it from orbit and rewrite the whole test case. However at that scope I think that no longer belongs to this PR.

Reply:
I like the part with a nuke but I guess that is for another millennium :).

Reply:
We can drop the commit for now and fix the issue in another PR.

Reply:
Let's do that!